core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -498,37 +515,41 @@ object AdminUtils extends Logging {
     changeEntityConfig(zkUtils, ConfigType.Topic, topic, configs)
   }
 
-  private def changeEntityConfig(zkUtils: ZkUtils, entityType: String, entityName: String, configs: Properties) {
+  private def changeEntityConfig(zkUtils: ZkUtils, rootEntityType: String, fullSanitizedEntityName: String, configs: Properties) {
+    val sanitizedEntityPath = rootEntityType + '/' + fullSanitizedEntityName
+    val entityConfigPath = getEntityConfigPath(rootEntityType, fullSanitizedEntityName)
```

Comment:
`sanitizedEntityPath` is `entityType/entityName`. This is the value stored in the JSON for config updates. `entityConfigPath` is the full path in Zookeeper. It is `/config/entityType/entityName`.